libeos-updater-util/tests/util.c

```diff
@@ -0,0 +1,148 @@
+  { "-3", SIGNED, -2, 2,  0,  TRUE },
+
+  { "1",  UNSIGNED, 0, 2, 1, FALSE },
+  { "0",  UNSIGNED, 0, 2, 0, FALSE },
+  { "2",  UNSIGNED, 0, 2, 2, FALSE },
+  { "3",  UNSIGNED, 0, 2, 0, TRUE },
+  { "-1", UNSIGNED, 0, 2, 0, TRUE },
+
+  { "",   SIGNED,   -2, 2, 0, TRUE },
+  { "",   UNSIGNED,  0, 2, 0, TRUE },
+  { "a",  SIGNED,   -2, 2, 0, TRUE },
+  { "a",  UNSIGNED,  0, 2, 0, TRUE },
+  { "1a", SIGNED,   -2, 2, 0, TRUE },
+  { "1a", UNSIGNED,  0, 2, 0, TRUE },
+};
+
+static void
+test_util_strtonum_usual (void)
+{
+  gsize idx;
+
+  for (idx = 0; idx < G_N_ELEMENTS(test_data); ++idx)
+    {
+      g_autoptr(GError) error = NULL;
+      TestData *data = &test_data[idx];
+      gboolean result;
+      gint value;
+
+      if (data->sign_type == SIGNED)
+        {
+          gint64 value64;
+          result = eos_string_to_signed (data->str,
+                                         data->min,
+                                         data->max,
+                                         &value64,
+                                         &error);
+          value = value64;
+          g_assert_cmpint (value, ==, value64);
+        }
+      else
+        {
+          guint64 value64;
+          result = eos_string_to_unsigned (data->str,
+                                           data->min,
+                                           data->max,
+                                           &value64,
+                                           &error);
+          value = value64;
+          g_assert_cmpint (value, ==, value64);
+        }
+
+      if (data->should_fail)
+        {
+          g_assert_false (result);
+          g_assert_false (error == NULL);
+        }
+      else
+        {
+          g_assert_true (result);
+          g_assert_true (error == NULL);
+          g_assert_cmpint (value, ==, data->expected);
+        }
+    }
+}
+
+static void
+test_util_strtonum_pathological (void)
+{
+  GError *error = NULL;
+  const gchar *crazy_high = "999999999999999999999999999999999999";
+  const gchar *crazy_low = "-999999999999999999999999999999999999";
+
+  g_assert_false (eos_string_to_unsigned (crazy_high, 0, G_MAXUINT64, NULL, &error));
+  g_assert_true (error != NULL);
+  g_clear_error (&error);
+  g_assert_false (eos_string_to_unsigned (crazy_low, 0, G_MAXUINT64, NULL, &error));
+  g_assert_true (error != NULL);
+  g_clear_error (&error);
+
+  g_assert_false (eos_string_to_signed (crazy_high, G_MININT64, G_MAXINT64, NULL, &error));
+  g_assert_true (error != NULL);
+  g_clear_error (&error);
+  g_assert_false (eos_string_to_signed (crazy_low, G_MININT64, G_MAXINT64, NULL, &error));
+  g_assert_true (error != NULL);
+  g_clear_error (&error);
+}
+
+int
+main (int   argc,
+      char *argv[])
+{
+  setlocale (LC_ALL, "");
+
+  g_test_init (&argc, &argv, NULL);
+
+  g_test_add_func ("/util/strtonum/usual", test_util_strtonum_usual);
+  g_test_add_func ("/util/strtonum/pathological", test_util_strtonum_pathological);
+  /* TODO: More */
```

Comment:
Ah, that's a copy-pasta from other file in the same directory. I think I'll extend it to mention maybe other functions from the `util.h` header.